includes/core-changes/serialization/5.0/options-honored-when-serializing-key-value-pairs.md

```diff
@@ -0,0 +1,89 @@
+
+```csharp
+var options = new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase };
+string json = @"{""key"":1,""value"":1}";
+// Throws JsonException.
+JsonSerializer.Deserialize<KeyValuePair<int, int>>(json, options);
+```
+
+Starting in .NET 5.0, the <xref:System.Text.Json.JsonSerializerOptions.PropertyNamingPolicy> and <xref:System.Text.Json.JsonSerializerOptions.PropertyNameCaseInsensitive> options are honored when deserializing using <xref:System.Text.Json.JsonSerializer>. For example, the following code snippet shows successful deserialization of lowercased <xref:System.Collections.Generic.KeyValuePair%602.Key> and <xref:System.Collections.Generic.KeyValuePair%602.Value> property names.
+
+```csharp
+var options = new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase };
+string json = @"{""key"":1,""value"":1}";
+
+KeyValuePair<int, int> kvp = JsonSerializer.Deserialize<KeyValuePair<int, int>>(json);
+Console.WriteLine(kvp.Key); // 1
+Console.WriteLine(kvp.Value); // 1
+```
+
+To accommodate payloads that were serialized with previous versions, "Key" and "Value" are special-cased to match when deserializing. Even though the <xref:System.Collections.Generic.KeyValuePair%602.Key> and <xref:System.Collections.Generic.KeyValuePair%602.Value> property names aren't camel-cased according to the in <xref:System.Text.Json.JsonSerializerOptions.PropertyNamingPolicy> option in the following code example, they deserialize successfully.
+
+```csharp
+var options = new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase };
+string json = @"{""Key"":1,""Value"":1}";
+
+KeyValuePair<int, int> kvp = JsonSerializer.Deserialize<KeyValuePair<int, int>>(json);
+Console.WriteLine(kvp.Key); // 1
+Console.WriteLine(kvp.Value); // 1
+```
+
+#### Version introduced
+
+5.0
+
+#### Reason for change
+
+Substantial customer feedback indicated that the <xref:System.Text.Json.JsonSerializerOptions.PropertyNamingPolicy> should be honored. For completeness, the <xref:System.Text.Json.JsonSerializerOptions.PropertyNameCaseInsensitive> and <xref:System.Text.Json.JsonSerializerOptions.Encoder> options are also honored, so that <xref:System.Collections.Generic.KeyValuePair%602> instances are treated the same as any other plain old CLR object (POCO).
+
+#### Recommended action
+
+If this change is disruptive to you, you can use a [custom converter](../../../../docs/standard/serialization/system-text-json-converters-how-to.md) that implements the desired semantics.
+
+#### Category
+
+Serialization
+
+#### Affected APIs
+
+- <xref:System.Text.Json.JsonSerializer.Serialize(System.Object,System.Type,System.Text.Json.JsonSerializerOptions)?displayProperty=fullName>
```

Comment:
Should I list other overloads of serialize and deserialize? Should I list the options as affected APIs?

Reply:
The change affects all the `Serialize`, `SerializeToUtf8Bytes`, `SerializeAsync`, `Deserialize`, and `DeserializeAsync` overloads on `JsonSerializer`. I'd expect something similar to what we had for the [deserialization constructors change](https://github.com/dotnet/docs/pull/21025/files#diff-c1fa0213bab7466b76f3fa016b9ccd1d8b7cdca9bd14075355c208852638d493R37-R38) here.

---

This change honors those options, but doesn't change their behavior. We shouldn't list them here.